### PR TITLE
basename: add -a, -s and -z flags

### PR DESCRIPTION
**Pull request by flyrry**
I noticed that some utilities are also using getopts or clap. Stuck with coreopts in this case to keep changes to a minimum. Appreciate any feedback.

### CODE REVIEW

File: src/basename/basename.rs [hunk 1]
```diff
@@ -47,23 +54,34 @@ pub fn uumain(args: Vec<String>) -> i32 {
...
+    let paths = if multiple_paths {
+        &matches.free[..]
+    } else {
+        &matches.free[0..1]
+    };
 
-    if matches.free.len() > 1 {
-        let suffix = matches.free[1].clone();
-        name = strip_suffix(name.as_ref(), suffix.as_ref());
+    for path in paths {
```
Comment (Arcterus) [inline src/basename/basename.rs]: I'd change this to:
```rust
let line_ending = if opt_z { "\0" } else { "\n" };
for path in paths {
    print!("{}{}", basename(&path, &suffix), line_ending);
}
```

**Comment by Arcterus** [comment]
Yeah, coreopts was kind of a stopgap.  I think we should migrate most things over to clap at some point.

**Comment by flyrry** [comment]
thanks for the feedback, updated

**Comment by Arcterus** [comment]
Thanks :+1:

**Pull request merged**